sdk/storage/azure_storage_blob/src/clients/blob_service_client.rs

```diff
@@ -52,14 +53,34 @@ impl BlobServiceClient {
             .per_try_policies
             .push(Arc::new(oauth_token_policy) as Arc<dyn Policy>);
 
-        let client = GeneratedBlobServiceClient::new(endpoint, credential, Some(options))?;
+        let client = GeneratedBlobServiceClient::new(endpoint, credential.clone(), Some(options))?;
 
         Ok(Self {
             endpoint: endpoint.parse()?,
+            credential,
             client,
         })
     }
 
+    /// Returns a new instance of BlobContainerClient.
+    ///
+    /// # Arguments
+    ///
+    /// * `container_name` - The name of the container.
+    /// * `options` - Optional configuration for the client.
+    pub fn blob_container_client(
+        &self,
+        container_name: String,
+        options: Option<BlobContainerClientOptions>,
+    ) -> Result<BlobContainerClient> {
+        BlobContainerClient::new(
+            self.endpoint().as_str(),
+            container_name,
+            self.credential.clone(),
+            options,
+        )
```

Comment:
Here and anywhere else: it'd be more efficient to just create the struct, as long as the fields are `pub(crate)` (effectively `internal` as in C#). As it is, you're passing a string for the endpoint just to convert it back to a URL. If you construct the struct, you can simply clone it:

```rust
BlobContainerClient {
  endpoint: self.endpoint.clone(),
  container_name,
  credential: credential.clone(),
  options,
}
```
This is only a minor perf improvement so I wouldn't go out of your way to make changes to make this work, but something to consider.
```